lib/email/receiver.rb

```diff
@@ -801,44 +801,60 @@ def embedded_email_raw
 
     def process_forwarded_email(destination, user)
       user ||= stage_from_user
-      embedded = Mail.new(embedded_email_raw)
-      email, display_name = parse_from_field(embedded)
-
-      return false if email.blank? || !email["@"]
-
-      raw = try_to_encode(embedded.decoded, "UTF-8").presence || embedded.to_s
-      title = embedded.subject.presence || subject
+      case SiteSetting.forwarded_emails_behaviour
+      when "create_replies"
+        forwarded_email_create_replies(destination, user)
+      when "quote"
+        forwarded_email_quote_forwarded(destination, user)
+      else
+        false
+      end
+    end
 
+    def forwarded_email_create_topic(destination, user, raw, title, date = nil, embedded_user = nil)
```

Comment:
I say given we have so many params ... let's name them all: 

`forwarded_email_create_topic(destination: , user: , raw:, title:, date: nil, embedded_user: nil)`

It makes the callers way clearer. I think once we reach about 3 params we should consider strongly naming params.

Reply:
I had though that there were too many arguments, and had considered passing them in as a hash, but hadn't realised this was possible (and so simple!)

I've added a commit doing this